### PR TITLE
 Fix NaNGuardMode to permit None variables

### PR DESCRIPTION
**Pull request by nouiz**
finish https://github.com/Theano/Theano/pull/4780

### CODE REVIEW

**Comment by nouiz** [comment]
jenkins test this

File: theano/compile/nanguardmode.py [hunk 1]
```diff
@@ -23,6 +23,37 @@
...
+    is_non_numeric : bool
+        `True` the value is non-numeric.
+
+    """
+    if isinstance(arr, theano.gof.type._cdata_type):
+        return True
+    elif isinstance(arr, np.random.mtrand.RandomState):
+        return True
+    elif var and getattr(var.tag, 'is_rng', False):
+        return False
```
Comment (lamblin) [inline theano/compile/nanguardmode.py]: Should be `True` here.

File: theano/compile/nanguardmode.py [hunk 2]
```diff
@@ -23,6 +23,37 @@
 logger = logging.getLogger("theano.compile.nanguardmode")
 
 
+def _non_numeric_value(arr, var):
```
Comment (lamblin) [inline theano/compile/nanguardmode.py]: I think it would be more clear not to invert the logic.

**Comment by nouiz** [comment]
both comment are done.

**Pull request merged**